Reason: port: code-mod hooks (sms_modhook.h): TMario::perform FLUDD model on Yoshi (BetterSunshineEngine)
--- a/src/Player/MarioMain.cpp
+++ b/src/Player/MarioMain.cpp
@@ -20,6 +20,10 @@
 #include <MSound/MSSetSound.hpp>
 #include <MSound/MSoundBGM.hpp>
 
+#ifdef TARGET_PC
+#include <sms_modhook.h>
+#endif
+
 static JGeometry::TVec3<f32> cDeformedTerrainCenter(0.0f, 5000.0f, 0.0f);
 
 TMario* gpMarioOriginal;
@@ -201,7 +205,18 @@
 			addDirty();
 			addDamageFog(graphics);
 
-			if (checkFlag(MARIO_FLAG_HAS_FLUDD))
+#ifdef TARGET_PC
+			// BetterSunshineEngine (isYoshiMaintainFluddModel, retail
+			// 0x8024D68C, with the word at 0x8024D690: cmpwi r3, 0): its
+			// result (it reads this from r31) replaces the FLUDD flag test.
+			if (SMS_MOD_SITE(0x8024D68C)
+			        ? (SMS_MOD_GPR(31, this),
+			           ((bool (*)())SMS_MOD_SITE(0x8024D68C))())
+			        :
+#else
+			if (
+#endif
+			    checkFlag(MARIO_FLAG_HAS_FLUDD))
 				MarioMainGetFludd(this)->perform(CUE_ENTRY, graphics);
 
 			entryModels(graphics);
